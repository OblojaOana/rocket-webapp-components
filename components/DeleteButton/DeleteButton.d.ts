export interface DeleteButtonProps {
    /**
    * The title of button.
    */
    title?: String;
    /**
     * Callback fired when a "click" event is detected.
     */
    onClick?: () => void
    /**
    * If true, the button will be disabled.
    */
    disabled?: Boolean
    /**
    * The color of button.
    * @default "themeNoBackground"
    */
    color?: "primary" |
            "info" |
            "theme" |
            "themeNoBackground" |
            "themeWithBackground" |
            "success" |
            "warning" |
            "danger" |
            "rose" |
            "white" |
            "simple" |
            "defaultNoBackground" |
            "primaryNoBackground" |
            "infoNoBackground" |
            "successNoBackground" |
            "warningNoBackground" |
            "dangerNoBackground" |
            "roseNoBackground"
    /**
      * The size of button.
      * @default "medium"
      */
    size?: "medium" | "small"
    /**
    * The fontSize of button's content.
    * @default "default"
    */
    fontSize?: "inherit" | "default" | "small" | "large"

}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/delete-button
 *
 * 
 */
export default function DeleteButton(props: DeleteButtonProps): JSX.Element;