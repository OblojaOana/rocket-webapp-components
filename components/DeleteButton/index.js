import DeleteButton from "../../core/DeleteButton";
export default DeleteButton;