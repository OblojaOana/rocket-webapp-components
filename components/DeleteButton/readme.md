##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import DeleteButton from '@bit/totalsoft_oss.react-mui.delete-button';

export default (
	<DeleteButton
		title="Delete"
	></DeleteButton>
)
 ```