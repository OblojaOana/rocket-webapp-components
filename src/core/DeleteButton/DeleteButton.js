import React from "react";
import PropTypes from "prop-types";
import DeleteIcon from '@material-ui/icons/Delete';
import CustomIconButton from '../IconButton';

const DeleteButton = ({ title, onClick, disabled, fontSize, ...rest }) => (
    <CustomIconButton aria-label="Delete" onClick={onClick} disabled={disabled} tooltip={title} {...rest}>
        <DeleteIcon fontSize={fontSize} />
    </CustomIconButton>
)

DeleteButton.defaultProps = {
    color: "themeNoBackground",
    fontSize: "default",
    size: "small"
};

DeleteButton.propTypes = {
    title: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground"
    ]),
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    fontSize: PropTypes.oneOf([
        "inherit",
        "default",
        "small",
        "large"
    ])
};

export default DeleteButton;