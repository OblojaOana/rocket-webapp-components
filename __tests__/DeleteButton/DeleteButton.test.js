import React from 'react';
import renderer from 'react-test-renderer';
import DeleteButton from '../../components/DeleteButton/DeleteButton';

describe('DeleteButton', () => {
    it('snapshot', () => {
        const component = renderer.create(<DeleteButton
            title="Delete"
            onClick={() => { }}
        ></DeleteButton>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});