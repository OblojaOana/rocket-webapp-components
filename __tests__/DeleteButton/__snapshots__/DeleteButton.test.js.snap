// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton snapshot 1`] = `
<button
  aria-describedby={null}
  aria-label="Delete"
  className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17 Mui-disabled Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  title="Delete"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
  </span>
</button>
`;
